oidc/provider.go

```diff
@@ -644,7 +650,7 @@ func (r *httpProviderConfigGetter) Get() (cfg ProviderConfig, err error) {
 
 	// The issuer value returned MUST be identical to the Issuer URL that was directly used to retrieve the configuration information.
 	// http://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationValidation
-	if !urlEqual(cfg.Issuer.String(), r.issuerURL) {
+	if !urlEqual(strings.TrimSuffix(cfg.Issuer.String(), "/"), r.issuerURL) {
```

Comment:
Why trim the suffix here? Shouldn't this be exactly equal? E.g. if r.issuerURL is "https://foo/auth/" then the returned issuer should be exactly "https://foo/auth/" as well.